src/coreclr/System.Private.CoreLib/src/System/Reflection/Metadata/RuntimeTypeMetadataUpdateHandler.cs

```diff
@@ -11,11 +11,16 @@ namespace System.Reflection.Metadata
     /// <summary>Metadata update handler used to clear a Type's reflection cache in response to a metadata update notification.</summary>
     internal static class RuntimeTypeMetadataUpdateHandler
     {
+        private static bool s_cacheCleared;
+        public static bool UpdateSupportedAndCacheCleared => MetadataUpdater.IsSupported && s_cacheCleared;
```

Comment:
Why is the IsSupported check needed here?  Is that to help with trimming?

If it's for trimming, ok.  Otherwise, s_cacheCleared would never be set to true if !IsSupported.

Reply:
It is to help with both trimming and with tiering compilation. 

`MetadataUpdater.IsSupported` is readonly and so the tiered JIT is able to dead-code eliminate the whole check and make this change zero perf impact when hot-reload is not supported.

Reply:
Makes sense.

Reply:
I wonder then if we're making this too complicated.  Should we just always use the slightly more expensive Equals/GetHashCode implementations if MetadataUpdater.IsSupported is true, and skip this cache clearing tracking?  I don't have a strong opinion, but there are plenty of other expenses when debugging, which is when MetadataUpdater.IsSupported would be true.

Reply:
Yes, I think it would make sense.

Reply:
If the cache never cleared the equals could be a bit faster, but there is not way to Reset it back, so after one hot reload it would always use slower method, so I am OK with removing this property